GUI improvements

taltamir opened the pull request:
GUI improvements:
* expose auto_getSteelOrgan setting
* auto_breakstone setting removed. mafia tracks it now
* auto_pvpEnable promoted from pre to any setting. can now be changed at any time.
* auto_alwaysGetSteelOrgan does not need to be converted to boolean just so that we can convert it back to a string and store it in auto_getSteelOrgan
* initialize settings when visiting GUI for the first time in an ascension. So we can immediately start changing post initialization settings.
* clearer wording in settings
* relay report 100% familiar status
* relay add buttons to change 100% familiar status if the run has not been started yet.
* initialize settings could fail to initialize settings over many ascensions if the tracker value of auto_doneInitialize was misconfigured. such as when you import settings from an old account to a new account. This is because it had an unnecessary `<=` where it should have been a `==`
* GUI state whether it is ok to change Post type settings
* remove depreciated auto_aftercore. mafia tracks it
* remove auto_getSteelOrgan from quests gui. it is shown in main gui
* auto_needLegs is POST not PRE
* auto_crackpotjar is POST not ANY
* various settings should be POST and were not
* remove depreciated auto_skipDesert
* remove depreciated auto_cubeItems
* remove depreciated auto_aboocount
* removed obsolete auto_dinseyGarbageMoney leftovers
* various other corrections on POST vs ANY vs PRE. Various other depreciated items removed from autoscend_quests.txt

* move autoscend_quests.txt to data folder
* changes in prep for rename of autoscend_quests.ash into autoscend_settings_extra.ash
* rename autoscend_quests.ash to autoscend_settings_extra.ash
* include backlink to return to main autoscend relay
* clearer text
* autoscend_settings_extra.ash switched to BUILD instead of tedious and prone to error manual formatting of datafile
** moved data to dat files in BUILD directory which compile into autoscend_settings_extra.txt
** import relay_autoscend.ash and use its functions instead of having duplicate but outdated functions
* void write_settings_key() created and used. moved existing code into it and eliminated its duplication across the two relay scripts
* auto_helpMeMafiaIsSuperBrokenAaah renamed to auto_inv_paranoia
* expose auto_disableFamiliarChanging in settings_extra

## How Has This Been Tested?

tested via visiting relay repeatedly with my various alterations at various game states. and clicking the new buttons to check their effects.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have based by pull request against the [beta branch](https://github.com/Loathing-Associates-Scripting-Society/autoscend/tree/beta) or have a good reason not to.

taltamir commented:
If someone knows how to do it. I would like to have a `run autoscend` button right underneath the `interrupt script` button.

taltamir commented:
TODO in followup PR:
1. switch auto_hedge from "fast" and "slow" to "true" and "false". maybe rename it too.
2. add a button at the top for "run autoscend" that when clicked runs autoscend. thus allowing autoscend to be launched from the relay GUI

Malibu-Stacey reviewed (changes requested):
The quest relay page contains internal tracking properties which are unlikely to provide any benefit to any user except for debugging. They should be kept on that page and we should move some other stuff that has been recently added to the main page there too (e,g.  `auto_doGalaktik`).

taltamir commented:
> The quest relay page contains internal tracking properties which are unlikely to provide any benefit to any user except for debugging. They should be kept on that page and we should move some other stuff that has been recently added to the main page there too (e,g. `auto_doGalaktik`).

Alright. I dropped the relevant commits.
I will instead work on making it more usable. such as moving its data file to the data directory and switching it from manual modification to being part of the build process of assemble file the same way our main gui works.

Also considering how many properties it has that are not quests I think we should rename it "additional settings" instead of quests

taltamir commented:
I finished moving the quests relay into the build method. switched over all the data, compiled it. and then switched how the script itself works. it had duplicate but outdated functions compared to relay_autoscend.ash
so I am having it import relay_autoscend.ash and use its functions (while deleting the duplicate outdated versions in quests). also renamed quests to autoscend_settings_extra.ash

taltamir commented on RELEASE/scripts/autoscend/iotms/mr2015.ash:
```diff
@@ -1189,10 +1189,6 @@ boolean LX_dinseylandfillFunbucks()
 	{
 		return false;
 	}
-	if(get_property("auto_dinseyGarbageMoney").to_int() == my_daycount())
-	{
-		return false;
-	}
```
this was meant to ensure we do not get money more than once a day. However it was not reset to 0 on ascension so it made it so we do not get money ever.
also we have since already added use of mafia tracking to handle it better via `_dinseyGarbageDisposed`
